### PR TITLE
storage(id)

### PR DESCRIPTION
**Pull request by Yborline**

### CODE REVIEW

File: src/js/locale-stor.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        const array = []
+    
+        if (watched) {
+
+            if (watched.includes(this.id)) {
+               return alert('такой фильм уже есть')
+            }
+
+            array.push(watched , this.id)
+            localStorage.setItem('watched', array)
```
Comment (Yurii97) [inline src/js/locale-stor.js]: ?

Comment (Yurii97) [reply src/js/locale-stor.js]: попробуй в масив запхати обєкт з даними { назва фільму , рік, ссилка на картинку, жанри(масив жанрів), рейтинг}

зданого обєкту зарендерим картку фільму

Comment (Yurii97) [reply src/js/locale-stor.js]: + в також id - щоб по ньому можна було знайти всю інфу по фільму(на модальне вікно)

**Review by Yurii97 — COMMENTED** [review]
Внести правки

**Review by Yurii97 — COMMENTED** [review]
+ добав метод класу який повертає масив обєктів

File: src/js/locale-stor.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                const pars = JSON.parse(watched)
+                array.push(...pars , this.obj)
+                setStorageWatched(array)
+        } else{
+                array.push(this.obj)
+                setStorageWatched(array)
+            }
+    }
+    getLocalStorage() {
+        const parseli = localStorage.getItem('watched')
```
Comment (Yurii97) [inline src/js/locale-stor.js:54]: даний метод має повертати parseli

File: src/js/locale-stor.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        this.getArray.push(JSON.parse(parseli))
```
Comment (Yurii97) [inline src/js/locale-stor.js]: це лишнє - перевір - ти масив пхаєш у масив - не потрібно

File: src/js/locale-stor.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+           this.getArray = [];
+    }
+
+    saveLocalStorage() {
+        const watched = localStorage.getItem('watched');
+     const objString = JSON.stringify(this.obj)
+        const array = []
+    
+        if (watched) { 
+            if (watched.includes(objString)) {
```
Comment (Yurii97) [inline src/js/locale-stor.js]: перевірка не працює - потрібно пофіксити

File: src/js/locale-stor.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+//     popularity: "100,2",
+//     genre: "Western",
+//     about: "Four of the West’s most infamous outlaws assemble to steal a huge stash of gold from the most corrupt settlement of the gold rush towns. But not all goes to plan one is killed and the other three escapes with bags of gold hide out in the abandoned gold mine where they happen across another gang of three – who themselves were planning to hit the very same bank! As tensions rise, things go from bad to worse as they realise the bags of gold are filled with lead... they’ve been double crossed – but by who and how?", img: "https://www.themoviedb.org/t/p/w300_and_h450_bestv2/ci5A9TPmNajMxt1L8p4KlZ76sc9.jpg"
+// }
+
+
+class LocalStorWatch {
+    constructor(obj) { 
+        this.obj = obj
+           this.getArray = [];
```
Comment (Yurii97) [inline src/js/locale-stor.js]: по коду не використовується

File: src/js/locale-stor.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+function setStorageWatched(array) {
+    localStorage.setItem('watched', JSON.stringify(array))
+
+}
+
+
+function setStorageQueue(array) {
+    localStorage.setItem('queue', JSON.stringify(array))
+
+}
```
Comment (Yurii97) [inline src/js/locale-stor.js]: Перевір чи працює мій варіант (пофікшений твій):

Comment (Yurii97) [reply src/js/locale-stor.js]: class LocalStorQueue {
    constructor() {   
    }
    saveLocalStorage(obj) {
        const queue = localStorage.getItem('queue');
        const array = []
        if (queue) { 
            if (queue.includes(JSON.stringify(obj))) {
            return alert('такой фильм уже есть')
            }
                const pars = JSON.parse(queue)
                array.push(...pars , obj)
                localStorage.setItem('queue', JSON.stringify(array))
        } else{
                array.push(obj)
                localStorage.setItem('queue', JSON.stringify(array))
            }
    }
    getLocalStorage() {
        const parseli = localStorage.getItem('queue')        
        return parseli;
    }
}

Comment (Yurii97) [reply src/js/locale-stor.js]: const obj1 = {
    name: "vaasya",
    vote: "8.3",
    votes: "12000000",
    popularity: "100,2",
    genre: "Western",
    about: "Four of the West’s most infamous outlaws assemble to steal a huge stash of gold from the most corrupt settlement of the gold rush towns. But not all goes to plan one is killed and the other three escapes with bags of gold hide out in the abandoned gold mine where they happen across another gang of three – who themselves were planning to hit the very same bank! As tensions rise, things go from bad to worse as they realise the bags of gold are filled with lead... they’ve been double crossed – but by who and how?", img: "https://www.themoviedb.org/t/p/w300_and_h450_bestv2/ci5A9TPmNajMxt1L8p4KlZ76sc9.jpg"
}
const obj2 = {
    name: "vaafya",
    vote: "8.3566",
    votes: "12000",
    popularity: "100,245",
    genre: "Western",
    about: "Four of the West’s most infamous outlaws assemble to steal a huge stash of gold from the most corrupt settlement of the gold rush towns. But not all goes to plan one is killed and the other three escapes with bags of gold hide out in the abandoned gold mine where they happen across another gang of three – who themselves were planning to hit the very same bank! As tensions rise, things go from bad to worse as they realise the bags of gold are filled with lead... they’ve been double crossed – but by who and how?", img: "https://www.themoviedb.org/t/p/w300_and_h450_bestv2/ci5A9TPmNajMxt1L8p4KlZ76sc9.jpg"
}
const obj3 = {
    name: "vaasya",
    vote: "8.3",
    votes: "12000000",
    popularity: "100,2",
    genre: "Western",
    about: "Four of the West’s most infamous outlaws assemble to steal a huge stash of gold from the most corrupt settlement of the gold rush towns. But not all goes to plan one is killed and the other three escapes with bags of gold hide out in the abandoned gold mine where they happen across another gang of three – who themselves were planning to hit the very same bank! As tensions rise, things go from bad to worse as they realise the bags of gold are filled with lead... they’ve been double crossed – but by who and how?", img: "https://www.themoviedb.org/t/p/w300_and_h450_bestv2/ci5A9TPmNajMxt1L8p4KlZ76sc9.jpg"
}

Comment (Yurii97) [reply src/js/locale-stor.js]: const localStoreQueue = new LocalStorQueue();

localStoreQueue.saveLocalStorage(obj1);
localStoreQueue.saveLocalStorage(obj2)
localStoreQueue.saveLocalStorage(obj3)
localStoreQueue.getLocalStorage();

File: src/partials-html/gallery.html [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <section>
 
   <div class="container">
-  
+
```
Comment (Yurii97) [inline src/partials-html/gallery.html:4]: не твій файл!

**Review by Yurii97 — COMMENTED** [review]
виправ - галерея не твій файл

**Review by Yurii97 — CHANGES_REQUESTED** [review]
ВИДАЛИТИ ЦЮ ГІЛКУ

**Comment by Vladosik12333** [comment]
New PR

**Pull request closed without merging**